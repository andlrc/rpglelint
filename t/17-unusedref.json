[
   {
      "column":7,
      "file":"t/17-unusedref.rpgle",
      "line":3,
      "message":"'b' defined but not used",
      "severity":"warning"
   }
]
