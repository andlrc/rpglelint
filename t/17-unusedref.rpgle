// rpglelint: -Wunused-variable
dcl-s a int(10);
dcl-s b int(10);
dcl-ds my_ds;
  subf1 int(10);
end-ds;

dcl-pr q;
end-pr;

dcl-ds my_ds2 likeds(my_ds);
dcl-ds my_ds3 likeds(my_ds) qualified;
dcl-ds my_ds4 likeds(my_ds) qualified;

dcl-ds my_ds_x;
  subf2 int(20);
end-ds;

dcl-ds my_ds_y template;
  subf3 int(20);
end-ds;

dcl-ds my_ds5 likeds(my_ds_x) qualified;

my_ds4.subf = a + subf1 + my_ds3.subf;

my_ds5.subf2 = 1;
