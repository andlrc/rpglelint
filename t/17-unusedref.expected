t/17-unusedref.rpgle:3:7: warning: 'b' defined but not used [-Wunused-variable]
 dcl-s b int(10);
       ^
